llvm/lib/Support/Compression.cpp

```diff
@@ -202,6 +207,58 @@ Error zstd::decompress(ArrayRef<uint8_t> Input,
   return E;
 }
 
+static void zstdContextFree(ZSTD_DCtx *Ctx) {
+  size_t RC = ::ZSTD_freeDCtx(Ctx);
+  assert(::ZSTD_isError(RC) == 0);
+}
+
+Error zstd::decompress(ArrayRef<uint8_t> Input,
+                       SmallVectorImpl<uint8_t> &Output) {
```

Comment:
1. What's the difference between this and `zstd::decompress`? Docs.
2. Document the function body. Having two loops is confusing.

Reply:
1. Added docs in `Compression.h`
2. Added comments explaining what's going on